app/Controller/CoGroupsController.php

```diff
@@ -227,24 +227,40 @@ function checkWriteDependencies($reqdata, $curdata = null) {
   function checkWriteFollowups($reqdata, $curdata = null, $origdata = null) {
     // Add the co person as owner/member of the new group, but only via HTTP
     
-    if(!$this->request->is('restful') && $this->action == 'add') {
-      $cos = $this->Session->read('Auth.User.cos');
-      $this->redirectTarget = array('action' => 'edit', $this->CoGroup->id);
-      
-      // Member of current CO? (Platform admin wouldn't be)
-      if(isset($cos) && isset($cos[ $this->cur_co['Co']['name'] ]['co_person_id'])) {
-        $a['CoGroupMember'] = array(
-          'co_group_id' => $this->CoGroup->id,
-          'co_person_id' => $this->Session->read('Auth.User.co_person_id'),
-          'owner' => true,
-          'member' => true
-        );
-        
-        if(!$this->CoGroup->CoGroupMember->save($a)) {
-          $this->Flash->set(_txt('er.gr.init'), array('key' => 'information'));
-          return false;
+    if(!$this->request->is('restful')) {
+
+      if($this->action == 'add') {
+        $cos = $this->Session->read('Auth.User.cos');
+        $this->redirectTarget = array('action' => 'edit', $this->CoGroup->id);
+
+        // Member of current CO? (Platform admin wouldn't be)
+        if (isset($cos) && isset($cos[$this->cur_co['Co']['name']]['co_person_id'])) {
+          $a['CoGroupMember'] = array(
+            'co_group_id' => $this->CoGroup->id,
+            'co_person_id' => $this->Session->read('Auth.User.co_person_id'),
+            'owner' => true,
+            'member' => true
+          );
+
+          if (!$this->CoGroup->CoGroupMember->save($a)) {
+            $this->Flash->set(_txt('er.gr.init'), array('key' => 'information'));
+            return false;
+          }
         }
       }
+
+      if($this->action == 'edit') {
+        // return to the index view with default filters in place after edit
+        $this->redirectTarget = array(
+          'controller' => 'co_groups',
+          'action' => 'index',
+          'co' => filter_var($this->cur_co['Co']['id'],FILTER_SANITIZE_SPECIAL_CHARS),
+          $this->CoGroup->id,
```

Comment:
This is confusing and non-standard... why does and `index` view take a record id?

Reply:
The record id was accidentally left in (bad copy paste, I suspect). Removed!

Reply:
Update committed as of ee3fa84